### PR TITLE
Introduce abstract security transport testcase

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This commit introduces an AbstractSimpleSecurityTransportTestCase for
security transports. This classes provides transport tests that are 
specific for security transports. Additionally, it fixes the tests referenced in 
#33285.

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/AbstractSimpleSecurityTransportTestCase.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+            OutputStreamStreamOutput stream = new OutputStreamStreamOutput(socket.getOutputStream());
+            stream.writeByte((byte) 'E');
+            stream.writeByte((byte) 'S');
+            stream.writeInt(-1);
+            stream.flush();
+
+            CountDownLatch renegotiationLatch = new CountDownLatch(1);
+            HandshakeCompletedListener secondListener = event -> renegotiationLatch.countDown();
+            socket.addHandshakeCompletedListener(secondListener);
```
Comment (original-brownbear) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/AbstractSimpleSecurityTransportTestCase.java]: Just a comment here:

I think this is only half the fix (adding the listener before initiating the handshake, the short read timeout of 50ms will still not suffice to trigger the now async handshaking reliably, you can try this out by setting less than 50ms for the socket timeout and getting the failure almost every time eventually) needed for this test. 
 see https://github.com/elastic/elasticsearch/pull/33943

=> that said, will review the rest here shortly :)

**Review by original-brownbear — APPROVED** [review]
LGTM :)

**Comment by tbrooks8** [comment]
@original-brownbear I re-applied Awaits fix and your can remove that in your PR.

**Review by s1monw — APPROVED** [review]
LGTM2

**Pull request merged**